src/modules/m_chanhistory.cpp

```diff
@@ -72,8 +72,8 @@ class HistoryMode : public ParamMode<HistoryMode, SimpleExtItem<HistoryList> >
 		}
 
 		unsigned int len = ConvToNum<unsigned int>(parameter.substr(0, colon));
-		unsigned int time = InspIRCd::Duration(duration);
-		if (len == 0 || (len > maxlines && IS_LOCAL(source)))
+		unsigned long time;
+		if (!InspIRCd::Duration(duration, time) || len == 0 || (len > maxlines && IS_LOCAL(source)))
```

Comment:
This causes a duration of 0s to be allowed which wasn't previously. It should be something like this:

```suggestion
		if (!InspIRCd::Duration(duration, time) ||  time == 0 || len == 0 || (len > maxlines && IS_LOCAL(source)))
```

Reply:
A duration of `0` is valid for chanhistory, there wasn't a check for it previously.